crates/optimism/flashblocks/src/service.rs

```diff
@@ -167,6 +172,22 @@ where
             },
         )))
     }
+
+    /// Takes out `current` [`PendingBlock`] if `state` is not preceding it.
+    fn on_new_tip<E: Display>(
+        &mut self,
+        state: Result<CanonStateNotification<N>, E>,
+    ) -> Option<PendingBlock<N>> {
+        let state = state.inspect_err(|e| tracing::error!("{e}")).ok()?;
+        let parent = self.current.as_ref().map(PendingBlock::parent_hash)?;
+        let latest = state.tip_checked().map(RecoveredBlock::hash)?;
+
+        if parent == latest {
```

Comment:
this can just be `self.current.as_ref()?.parent_has() == state.tip_checked()?.hash()`